internal/funcspec/spec.go

```diff
@@ -22,6 +23,11 @@ func Spec(fn interface{}, args ...argmapper.Arg) (*pb.FuncSpec, error) {
 
 	filterProto := argmapper.FilterType(protoMessageType)
 
+	// Outparameters do not need to be supplied by core, and should
+	// be omitted from the advertised function spec.
+	// TODO(izaak): If and when we have more outparameter inputs, we should make this more generic.
+	filterDeclaredResourcesOutParam := argmapper.FilterType(reflect.TypeOf(&component.DeclaredResourcesResp{}))
```

Comment:
@mitchellh I couldn't get this to work with a generic interface type - I think I would have to make a new "Outparameter" interface with some unique dummy function (maybe "isOutparameter"), and then have DeclaredResourcesResp implement that function? I'm open to other ideas. We can probably polish this even more though if we end up using the pattern again.

Reply:
@izaaklauer yeah, that's the way to do it. Basically have:
```
type OutParam interface {
   IsOutParam()
}
```

then, on DeclRes: 

```
func (d *DeclRes) IsOutParam() {}
```

Really just using interfaces as ways to add types to a union type.

Reply:
Ha ok, that felt weird to me but I guess i'm on board.